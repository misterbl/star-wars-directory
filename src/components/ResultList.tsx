import React, { FunctionComponent } from "react";
import CharacterCard from "./CharacterCard";
import FilmCard from "./FilmCard";
import { IResultList } from "./ResultList.d";
import { ICharacter, IFilm } from "../state";

const ResultList: FunctionComponent<IResultList> = ({
  list,
  type,
  children
}: IResultList) => (
  <>
    <div className="grey-line" />
    <h3 className="text-white my-3 text-center">{`${type}:`}</h3>
    {children}
    <div className="grey-line mt-3" />
    <div className="p-2 mt-3">
      {list.map(element => {
        if (type === "Characters") {
          const character = element as ICharacter;
          return <CharacterCard key={character.name} character={character} />;
        }
        const film = element as IFilm;
        return <FilmCard key={film.title} film={film} />;
      })}
    </div>
  </>
);

export default ResultList;
