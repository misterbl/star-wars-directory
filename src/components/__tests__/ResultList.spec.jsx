/* eslint-disable no-undef */
import * as React from "react";
import { shallow } from "enzyme";
import ResultList from "../ResultList";

describe("ResultList", () => {
  const props = {
    list: ["element1", "element2"],
    type: "type"
  };
  const wrapper = shallow(<ResultList {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
