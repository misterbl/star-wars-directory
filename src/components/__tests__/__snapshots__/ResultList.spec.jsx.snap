// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultList
    list={
      Array [
        "element1",
        "element2",
      ]
    }
    type="type"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="grey-line"
        />,
        <h3
          className="text-white my-3"
        >
          type matching your search:
        </h3>,
        undefined,
        <div
          className="grey-line mt-3"
        />,
        <div
          className="p-2 mt-3"
        >
          <withRouter(Connect(FilmCard))
            film="element1"
          />
          <withRouter(Connect(FilmCard))
            film="element2"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "grey-line",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "type matching your search:",
          "className": "text-white my-3",
        },
        "ref": null,
        "rendered": "type matching your search:",
        "type": "h3",
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "grey-line mt-3",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Connect(FilmCard))
              film="element1"
            />,
            <withRouter(Connect(FilmCard))
              film="element2"
            />,
          ],
          "className": "p-2 mt-3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "film": "element1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "film": "element2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="grey-line"
          />,
          <h3
            className="text-white my-3"
          >
            type matching your search:
          </h3>,
          undefined,
          <div
            className="grey-line mt-3"
          />,
          <div
            className="p-2 mt-3"
          >
            <withRouter(Connect(FilmCard))
              film="element1"
            />
            <withRouter(Connect(FilmCard))
              film="element2"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "grey-line",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "type matching your search:",
            "className": "text-white my-3",
          },
          "ref": null,
          "rendered": "type matching your search:",
          "type": "h3",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "grey-line mt-3",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(Connect(FilmCard))
                film="element1"
              />,
              <withRouter(Connect(FilmCard))
                film="element2"
              />,
            ],
            "className": "p-2 mt-3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "film": "element1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "film": "element2",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
